### ISSUE TITLE
Add aggregate argument to batch_calc_ES().

### ISSUE THREAD

**Issue by jepusto** [jepusto/SingleCaseES#35]
Add aggregate argument to batch_calc_ES(), which would be a set of one or more grouping_vars that would be used to identify unique data series, but then aggregated over to arrive at a single effect size estimate. The result would be akin to something like
```
batch_calc_ES(..., grouping_vars = c(A, B, C), aggregate = c(D, E), format = "long") %>%
  group_by(A, B, C, ES) %>%
  summarise(
    Est = sum(Est / SE^2) / sum(1 / SE^2),
    SE = sqrt(1 / sum(1 / SE^2)),
    CI_L = ...,
    CI_U = ....,
  ) %>%
  wide_format_if_need_be()
```

One outstanding problem here is what to do with effect size estimates that do not have known SEs. Maybe choose a sensible but ad hoc weighting scheme. Might also be good to allow for equal weighting.

**Comment by jepusto** [comment]
Need to provide options for what weighting scheme to use, something like:

```
batch_calc_ES(..., grouping_vars = c(A, B, C), aggregate = c(D, E), weighting = "1/V", format = "long") %>%
  group_by(A, B, C, ES) %>%
  summarise(
    Est = sum(Est / SE^2) / sum(1 / SE^2),
    SE = sqrt(1 / sum(1 / SE^2)),
    CI_L = ...,
    CI_U = ....,
  ) %>%
  wide_format_if_need_be()
```

**Comment by jepusto** [comment]
As a first pass, the `weighting` argument should take values `"1/V"` or `"equal"`. 

Further options might be `"1/nA"`, `"1/nB"`, `"1/nA + 1/nB"`, or `"1 / (nA nB)"`.

**Comment by jepusto** [comment]
That was my first instinct too, but given our main user-base, I think just using a fixed set of options (denoted by character strings) might be safer and more accessible. And probably easier.

**Comment by DMSwan** [comment]
Having the option of passing a function would be pretty nice, though! It might be reasonable to check the class of the passed object and use a set of default options with a character string and something else if you pass it a function.

Or maybe just a different argument, like weighting_func which overrides the weighting string whenever it's non-null.

**Comment by jepusto** [comment]
Resolved with commit 23bae00d53e0a00f604c374c3321ed0de2d99851
